modules/curl/8.8.0/patches/add_build_file.patch

```diff
@@ -0,0 +1,307 @@
+diff --git a/BUILD.bazel b/BUILD.bazel
+new file mode 100644
+index 000000000..0eaa6df71
+--- /dev/null
++++ b/BUILD.bazel
+@@ -0,0 +1,301 @@
++# Copyright The OpenTelemetry Authors
++# SPDX-License-Identifier: Apache-2.0
++
++load("@bazel_skylib//rules:common_settings.bzl", "bool_flag", "string_flag")
++
++package(features = ["no_copts_tokenization"])
++
++bool_flag(
++    name = "use_mbedtls",
++    build_setting_default = False,
++)
++
++config_setting(
++    name = "use_mbedtls_setting",
++    flag_values = {":use_mbedtls": "true"},
++)
++
++bool_flag(
++    name = "http_only",
++    build_setting_default = True,
++)
++
++config_setting(
++    name = "http_only_setting",
++    flag_values = {":http_only": "true"},
++)
++
++# Curl's default CA path depends on the OS. If CURL_CA_BUNDLE/CURL_CA_PATH are not set,
```

Comment:
i think we should provide some defaults here. i started on this here https://github.com/bazelbuild/bazel-central-registry/pull/1943/files#diff-400335528b8cf846395d95f2877a3ff732563a63b7b0ef2be7bac192532e78acR627 actually not even allowing overrides. I think overrides are probably fine but this kinda locks us into a API

Reply:
Something similar was my first approach, in https://github.com/bazelbuild/bazel-central-registry/pull/2233/commits/16dfd0d6bd7eeb3b274844912e6b399eb6fb5b7d.

I think it would be OK to start small and build this out iteratively.

It would also be great to allow the user to bring-your-own hermetic cabundle although admittedly I haven't actually tried that yet.

Reply:
i think the defaults i linked are probably fine to start with so it "just works" in the common case on the explicitly supported OS's